backend/src/auth/auth.controller.ts

```diff
@@ -1,23 +1,107 @@
-import { Body, Controller, Post, Request, UseGuards } from '@nestjs/common';
+import { Body, Controller, Get, Post, Req, Res, UseGuards } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { AppConfigService } from 'src/config/config.service';
+import { NovuService } from 'src/novu/novu.service';
+import { UserService } from 'src/user/user.service';
 import { AuthService } from './auth.service';
 import { SignupDto } from './dto/signup.dto';
+import { JwtRefreshAuthGuard } from './guards/jwt-refresh.guard';
+import { JwtAuthGuard } from './guards/jwt.guard';
 import { LocalAuthGuard } from './guards/local.guard';
+import { VerifyAuthGuard } from './guards/verify.guard';
+import { UserContext } from './interfaces/user-context';
 
 @Controller({
   path: 'auth',
   version: '1',
 })
 export class AuthController {
-  constructor(private readonly authService: AuthService) {}
+  constructor(
+    private readonly authService: AuthService,
+    private readonly novuService: NovuService,
+    private readonly userService: UserService,
+    private readonly appConfigService: AppConfigService
+  ) {}
 
-  @Post('/login')
   @UseGuards(LocalAuthGuard)
-  async login(@Request() req: any) {
-    return this.authService.login(req.user);
+  @Post('/login')
+  async login(@Req() req: Request, @Res() res: Response) {
+    const tokens = await this.authService.login(req.user as UserContext);
+    const domain = process.env.NODE_ENV === 'production' ? '.reduced.to' : 'localhost';
+    res.cookie('accessToken', tokens.accessToken, {
+      expires: new Date(new Date().getTime() + 15 * 60 * 1000), //15 min
```

Comment:
Time here is 15 minutes, I think it should be 5 minutes as we did with the JWT token